src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -530,6 +540,29 @@ public Task DeleteAsync(HashHeightPair newTip, List<uint256> hashes)
             return task;
         }
 
+        /// <inheritdoc />
+        public Task DeleteBlocksAsync(List<uint256> hashes)
+        {
+            Guard.NotNull(hashes, nameof(hashes));
+
+            Task task = Task.Run(() =>
+            {
+                using (DBreeze.Transactions.Transaction transaction = this.DBreeze.GetTransaction())
+                {
+                    transaction.SynchronizeTables(BlockRepository.BlockTableName, BlockRepository.CommonTableName, BlockRepository.TransactionTableName);
+                    transaction.ValuesLazyLoadingIsOn = false;
+
+                    List<Block> blocks = this.GetBlocksFromHashes(transaction, hashes);
+
+                    this.OnDeleteBlocks(transaction, blocks.Where(b => b != null).ToList());
```

Comment:
Rename `OnDeleteBlocks` to `DeleteBlocksInternal` or smth. Same with txes. 
When methods start's with `On` it's a callback from an event. In this case it's not a callback

Reply:
This is changes that is not related to the PR. We can do later.